Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    ValueStringFilter::ValueStringFilter()
+        : RowFilterAdapter()
+    {
+    }
+
+    void ValueStringFilter::SetFilterString(AZStd::string filterString)
+    {
+        AZStd::to_lower(filterString.begin(), filterString.end());
+        if (m_filterString != filterString)
+        {
+            m_filterString = AZStd::move(filterString);
+
+            if (m_filterActive)
+            {
+                if (m_filterString.empty())
+                {
+                    SetFilterActive(false);
+                }
+                else
+                {
+                    InvalidateFilter();
+                }
+            }
+            else if (!m_filterString.empty())
+            {
+                SetFilterActive(true);
+            }
+        }
+    }
+
+    RowFilterAdapter::MatchInfoNode* ValueStringFilter::NewMatchInfoNode() const
+    {
+        return new StringMatchNode();
+    }
+
+    void ValueStringFilter::CacheDomInfoForNode(const Dom::Value& domValue, MatchInfoNode* matchNode) const
+    {
+        auto actualNode = static_cast<StringMatchNode*>(matchNode);
+        const bool nodeIsRow = IsRow(domValue);
+        AZ_Assert(nodeIsRow, "Only row nodes should be cached by a RowFilterAdapter");
+        if (nodeIsRow)
+        {
+            actualNode->m_matchableDomTerms.clear();
+            for (auto childIter = domValue.ArrayBegin(), endIter = domValue.ArrayEnd(); childIter != endIter; ++childIter)
+            {
+                auto& currChild = *childIter;
+                if (currChild.IsNode())
+                {
+                    auto childName = currChild.GetNodeName();
+                    if (childName != Dpe::GetNodeName<Dpe::Nodes::Row>()) // don't cache child rows, they have they're own entries
+                    {
+                        static const Name valueName = Name::FromStringLiteral("Value", AZ::Interface<AZ::NameDictionary>::Get());
```

Comment:
Can we use `PointerValueFieldName` to reduce duplication?

Reply:
I don't think that will work, because this is capital-V "Value", but I agree we should be using constants here. I think I might be able to extract his from the static strings in PropertyEditorNodes.h... stay tuned

Reply:
now extracting these from PropertyEditor declarations within PropertyEditorNodes.h. Let me know if that doesn't work for you...